.github/workflows/ci.yaml

```diff
@@ -0,0 +1,185 @@
+name: EC2 Metadata Mock CI and Release
+
+on: [push, pull_request, workflow_dispatch]
+
+env:
+  DEFAULT_GO_VERSION: ^1.15
+  IS_RELEASE: ${{ github.event_name == 'push' && contains(github.ref, 'refs/tags/') }}
+  IS_PUSH: ${{ github.event_name == 'push' }}
+  # Skip running this stage when updating release versions as part of release prep
+  # as the new Docker image wouldn't be available yet and previous commits should have already been tested.
+  # Since the release prep commit will be the release commit, always run this stage when a new tag is pushed, as part of a release.
+  SHOULD_TEST_HELM: ${{ !contains(github.event.head_commit.message, 'Skip Helm E2E Tests') || (github.event_name == 'push' && contains(github.ref, 'refs/tags/')) }}
+  GITHUB_USERNAME: ${{ secrets.EC2_BOT_GITHUB_USERNAME }}
+  GITHUB_TOKEN: ${{ secrets.EC2_BOT_GITHUB_TOKEN }}
+  DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+  DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
+  DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+
+jobs:
+
+  buildAndTest:
+    name: Build and Test
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.DEFAULT_GO_VERSION }}
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build
+      run: make build
+
+    - name: Unit Tests
+      run: make unit-test
+    
+    - name: Lints
+      run: make spellcheck shellcheck
+
+    - name: Go Report Card Tests
+      run: make go-report-card-test
+
+    - name: Readme Length Validator
+      run: make validate-readme
+    
+    - name: Brew Sync Dry run
+      run: make homebrew-sync-dry-run
+
+    - name: License Test
+      if: ${{ env.IS_PUSH == true }}
+      run: make license-test
+
+    - name: E2E Tests
+      run: make e2e-test
+
+    - name: Mock IP Count Test
+      run: make helm-mock-ip-count-test
+
+    - name: Build Release Assets
+      run: make build-release-assets
+
+    - name: Build Docker Images Linux
+      run: make build-docker-images-linux
+
+  buildWindows:
+    name: Build Docker Images Windows
+    runs-on: windows-2019
+    steps:
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.DEFAULT_GO_VERSION }}
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Build Docker Images Windows
+      run: choco install make && RefreshEnv.cmd && make build-docker-images-windows
+    
+  release:
```

Comment:
Can do the [If at the job level](https://github.blog/changelog/2019-10-01-github-actions-new-workflow-syntax-features/) so we don't need it for each step

Reply:
ohhhhhhh I didn't see that! I'll change